### PR TITLE
Fixed code to pass some tests in TopicCreateSuite, CryptoDeleteSuite, CryptoUpdateSuite and CryptoApproveAllowanceSuite

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Fixed code to enable some tests in:
 
TopicCreateSuite - all consensus services hapitests are passed by now;
CryptoDeleteSuite - 1 hapitest pending on #8463; 
CryptoUpdateSuite - 1 hapitest pending on #8463, another one pending on ContractUpdateHandler.handle() to be implemented;

CryptoApproveAllowanceSuite, passed tests are below
- [x] emptyAllowancesRejected
- [x] canHaveMultipleOwners
- [x] validatesSerialNums
- [x] serialsInAscendingOrder
- [x] cannotHaveMultipleAllowedSpendersForTheSameNFTSerial
- [x] approveForAllDoesNotSetExplicitNFTSpender
- [X] canGrantNftAllowancesWithTreasuryOwner
- [x] transferringMissingNftViaApprovalFailsWithInvalidNftId 

completed part of #8045.

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java [hunk 1]
```diff
@@ -159,9 +159,10 @@ private void validateCommon(
      */
     private void validateCounts(
             final int nftCount, final long fungibleCount, final boolean nftsAreEnabled, final long maxBatchSize) {
-        if (fungibleCount > 0) {
-            validateTrue(fungibleCount <= maxBatchSize, BATCH_SIZE_LIMIT_EXCEEDED);
-        }
+        // commented out below, checked with Michael T, nfts.maxBatchSizeMint is not for fungibleCount.
```
Comment (netopyr) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java:162]: If this is not needed, we should remove the code entirely.

**Review by netopyr — APPROVED** [review]
LGTM

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenSupplyChangeOpsValidator.java:162]: I left this for @Neeharika-Sompalli to review when she is back.

**Review by kimbor — APPROVED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoApproveAllowanceSuite.java [hunk 2]
```diff
@@ -532,6 +538,7 @@ private HapiSpec canGrantNftAllowancesWithTreasuryOwner() {
                                 .logged());
     }
 
+    //    @HapiTest
```
[the remark this answers is not in the document]
Comment (iwsimon) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoApproveAllowanceSuite.java:541]: Those tests are still failing.

**Review by netopyr — APPROVED** [review]
LGTM

**Pull request merged**